### DISCUSSION TITLE
PHP versions and packages

### DISCUSSION THREAD

**Discussion by p7s1-wolfgang-dworzak** [amazonlinux/amazon-linux-2022#130] [General]
As far a I see from the the AL2022 preview the available version of PHP is 7.4, but for this version security updates will be discontinued already in November 2022 according to https://www.php.net/supported-versions.php.

On AL2 there was already PHP 8 available via `amazon-linux-extras`, but that will be not an option anymore on AL2022 as AWS announced that there will be no `amazon-linux-extras` anymore. But there are no other namespace packages for PHP as well.

So my question is, if it is planned to provide PHP 8 also soon, because at https://docs.aws.amazon.com/linux/al2022/release-notes/planned-changes.html nothing is listed.

The second question is regarding additional PHP packages. On AL2 there are more PHP 7.4. like `php-pear`, which are not available in AL2022. Is there a reason for it?

**Comment by stewartsmith** [comment]
Yes, we plan to have PHP8 soon, and won't ship 7.4.

**Reply by p7s1-wolfgang-dworzak** [reply]
Thanks for the answer, sounds good.

But what about the provided packages, will the amount of PHP packages also increase to cover the most common ones like the mentioned `php-pear`? We anyway need to build some custom rpms, which depend on several packages like the mentioned one.

**Comment by lphilps** [comment]
I had the same PHP related issue with the AL2022 Beta, but was pleased to see that in the first release candidate php 7.4 has been replace by a newer version, PHP 8.0. However, I, at least, am still facing some issues:

- PHP 8.0 is only under active support for another 90 days (26 Nov 2022) so I was pretty surprised to to see PHP 8.0 in the RC and not PHP 8.1.  Is there a plan to make v8.1 available in the near future?
- The PHP 8.0 version that is included is missing the "sodium" extension, which, to the best of my understanding, can't even be built via pecl as of php 7.2?  It needs to be configured as part of the PHP build itself.
- There are also no pre-compiled PECL extensions, of which memcached and redis would be the most essential.
- Theoretically, the memcache extension could be built using pecl, but that requires that "libmemcached" be installed first and I can't find any way to install that package on AL 2022.

Any information would be very much appreciated.

**Comment by KenAir** [comment]
I am also looking on without amazon-linux-extras, how can I install PHP extension, like php-zip, not available in AL 2022 but could install in AL2 via amazon-linux-extras.

**Comment by stewartsmith** [comment]
We currently support PHP 8.1 and 8.2 in Amazon Linux 2023.

Requests for changes are being tracked over in https://github.com/amazonlinux/amazon-linux-2023/issues?q=is%3Aissue+is%3Aopen+php
